sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/AmqpMessageConverterTests.cs

```diff
@@ -710,5 +713,323 @@ public void CreateBatchFromMessagesWithMultipleEventsAssignsThePartitionKeyToBod
         }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpMessageConverter.CreateEventFromMessage" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void CreateEventFromMessageValidatesTheSource()
+        {
+            var converter = new AmqpMessageConverter();
+            Assert.That(() => converter.CreateEventFromMessage(null), Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpMessageConverter.CreateEventFromMessage" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void CreateEventFromMessagePopulatesTheBody()
+        {
+            var body = new byte[] { 0x11, 0x22, 0x33 };
+
+            using var bodyStream = new MemoryStream(body, false);
+            using var message = AmqpMessage.Create(bodyStream, true);
+
+            var converter = new AmqpMessageConverter();
+            var eventData = converter.CreateEventFromMessage(message);
+
+            Assert.That(eventData, Is.Not.Null, "The event should have been created.");
+            Assert.That(eventData.Body, Is.Not.Null, "The event should have a body.");
+            Assert.That(eventData.Body.ToArray(), Is.EqualTo(body), "The body contents should match");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpMessageConverter.CreateEventFromMessage" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void CreateEventFromMessagePopulatesSimpleApplicationProperties()
+        {
+            var propertyValues = new object[]
+            {
+                (byte)0x22,
+                (sbyte)0x11,
+                (Int16)5,
+                (Int32)27,
+                (Int64)1122334,
+                (UInt16)12,
+                (UInt32)24,
+                (UInt64)9955,
+                (Single)4.3,
+                (Double)3.4,
+                (Decimal)7.893,
+                Guid.NewGuid(),
+                DateTime.Parse("2015-10-27T12:00:00Z"),
+                true,
+                'x',
+                "hello"
+            };
+
+            var applicationProperties = propertyValues.ToDictionary(value => $"{ value.GetType().Name }Property", value => value);
+
+            using var bodyStream = new MemoryStream(new byte[] { 0x11, 0x22, 0x33 }, false);
+            using var message = AmqpMessage.Create(bodyStream, true);
+
+            foreach (var pair in applicationProperties)
+            {
+                message.ApplicationProperties.Map.Add(pair.Key, pair.Value);
+            }
+
+            var converter = new AmqpMessageConverter();
+            var eventData = converter.CreateEventFromMessage(message);
+
+            Assert.That(eventData, Is.Not.Null, "The event should have been created.");
+            Assert.That(eventData.Body, Is.Not.Null, "The event should a body.");
```

Comment:
Missing word: the event should **have** a body.

There are other occurrences in the same file.